services/community-site/src/pages/Staking/NodeProfilePage.tsx

```diff
@@ -23,65 +22,60 @@ import { useAllValidators } from '../../services/useAllValidators';
 import NodeProfilePageSkeleton from './Screen/NodeProfilePageSkeleton';
 import { useLoading } from '../../services/useLoading';
 import Nickname from '../../components/Nickname/Nickname';
+import { DelegationGQL } from '../../interfaces/Delegation';
+import { BarFlex } from './BarFlex';
 
-// interface BarFlexProps {
-//   communityDelegated: number;
-//   selfDelegated: number;
-//   availableDelegation: number;
-// }
+function calculateDelegationSpread(validator: Validator | null, delegations: DelegationGQL[]) {
+  const delegationPossible =
+    validator?.delegation
+      .add(validator?.availableForDelegation)
+      .div(BigNumber.from('10').pow(BigNumber.from('18')))
+      ?.toNumber() || 1;
 
-// const BarFlex = ({ communityDelegated, selfDelegated, availableDelegation }: BarFlexProps) => {
-//   let cdParsed = communityDelegated;
-//   let sdParsed = selfDelegated;
-//   let adParsed = availableDelegation;
-//   if (cdParsed + sdParsed + adParsed > 100) {
-//     /**
-//      * Highest number = 100 - sum of lowest numbers
-//      */
-//     if (cdParsed >= sdParsed && cdParsed >= adParsed) cdParsed = 100 - (sdParsed + adParsed);
-//     if (sdParsed >= cdParsed && sdParsed >= adParsed) sdParsed = 100 - (cdParsed + adParsed);
-//     if (adParsed >= cdParsed && adParsed >= sdParsed) adParsed = 100 - (sdParsed + cdParsed);
-//   }
-//   return (
-//     <div className="barFlex">
-//       <div
-//         className="percentageAmount"
-//         style={{
-//           width: `${cdParsed}%`,
-//         }}
-//       >
-//         <div className="barPercentage" style={{ background: '#15AC5B' }} />
-//       </div>
-//       <div
-//         className="percentageAmount"
-//         style={{
-//           width: `${sdParsed}%`,
-//         }}
-//       >
-//         <div className="barPercentage" style={{ background: '#8E8E8E' }} />
-//       </div>
-//       <div
-//         className="percentageAmount"
-//         style={{
-//           width: `${adParsed}%`,
-//         }}
-//       >
-//         <div className="barPercentage" style={{ background: '#48BDFF' }} />
-//       </div>
-//     </div>
-//   );
-// };
+  const _ownDelegation = delegations.find(
+    (d) => d.delegator.toLowerCase() === validator?.owner?.toLowerCase(),
+  );
+  const ownDelegation = _ownDelegation
+    ? BigNumber.from(_ownDelegation.amount)
+        .div(BigNumber.from('10').pow(BigNumber.from('18')))
+        ?.toNumber()
+    : 0;
+
+  const availableForDelegation = parseFloat(
+    validator?.availableForDelegation
+      ?.div(BigNumber.from('10').pow(BigNumber.from('18')))
+      ?.toString() || '0',
+  );
+
+  const validatorDelegation = parseFloat(
+    validator?.delegation?.div(BigNumber.from('10').pow(BigNumber.from('18')))?.toString() || '1',
+  );
+  const communityDelegatedTotal = validatorDelegation - ownDelegation;
```

Comment:
What happends if the `validator?.delegation?` is null? The parseFloat will return `NaN` so this substraction will throw an error, right? Might be good to test it and handle it somehow.